R/Ameriflux_data_tables.R

```diff
@@ -0,0 +1,562 @@
+##############################################################################
+# Adam Reevesman
+# March 19, 2018
+
+# This script defines the function "attribute_table" which
+# generates an attribute table for any one of the csvs in the Ameriflux data at
+# https://drive.google.com/drive/folders/1g0TuYt1cJEuRxe5tXoUjJ1pSneNFbPXN
+
+# The first step is to read in each of the csvs and my modified version of
+# the attribute metadata from
+# https://docs.google.com/spreadsheets/d/1K-C7DahQ80KifW1NAbFaU-VusLZe4eAPHqhBl4t_y9s/edit#gid=1340082892
+
+# the attribute metadata is read into a dataframe as definitions
+# the csvs are read in as dataframes called data1, data2, ... , data12
+
+# The function "attribute_table" takes in 
+# the data dataframe and definitions dataframe to produce an attribute table for the data
+# It calls indivual functions that define each column of the attribute table
+
+# The resulting attribute table will not be perfect...
+# The measurement scales, number types, and missing value information will need
+# to be adjusted manually
+# Any custom units that exist in the "unit" column will need to be defined
+
+# An example of the implimentation exists at the very end of this script
+
+##############################################################################
+
+library(stringr)
+
+
+definitions <- read.csv('/home/reevesman/Ameriflux/attribute_function/definitions.csv',
+                        stringsAsFactors = F)
+
+data1 <- read.csv('/home/reevesman/Ameriflux/AMF_US-Ivo/AMF_US-Ivo_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data2 <- read.csv('/home/reevesman/Ameriflux/AMF_US-ICt/AMF_US-ICt_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
```

Comment:
R is weird in that it lets you abbreviate TRUE and FALSE as T/F.  It's generally considered best practice to spell these out to increase readability.

Reply:
When you're making changes you can split up your commits.  For instance in one commit you can just change all the T/Fs to TRUE/FALSE, and just make that commit something like "updated T and F syntax"